maps/src/main/java/org/odk/collect/maps/layers/OfflineMapLayersPicker.kt

```diff
@@ -74,10 +76,6 @@ class OfflineMapLayersPicker(
     private val getLayers = registerForActivityResult(ActivityResultContracts.GetMultipleContents(), registry) { uris ->
         if (uris.isNotEmpty()) {
             sharedViewModel.loadLayersToImport(uris, requireContext())
-            DialogFragmentUtils.showIfNotShowing(
```

Comment:
I feel like this would all be a bit more contained, if `OfflineMapLayersImporter` handles the warning cases? `layersToImport` could still be a `Consumable`, but it wouldn't care about if it had consumed or not for setting the adapter. In that world, there's no changes in `OfflineMapLayersPicker` and in `OfflineMapLayersImporter` we just get something like:

```kotlin
viewModel.layersToImport.observe(this) { layersToImport ->
    val adapter = OfflineMapLayersImporterAdapter(layersToImport.value.layers)
    binding.layers.setAdapter(adapter)

    if (layersToImport.isConsumed()) {
        // Show warning dialog if it make sense

        layersToImport.consume()
    }
}
```

For me that keeps the navigation simpler. What do you think?

Reply:
But `OfflineMapLayersImporter` should not be displayed at all if all the selected files are not supported right?

Reply:
Ok done.